src/replication.c

```diff
@@ -4154,6 +4156,13 @@ void replicationCachePrimary(client *c) {
     c->bufpos = 0;
     resetClient(c);
 
+    /* Reset the primary IO state. */
+    c->nwritten = 0;
+    c->nread = 0;
+    c->io_read_state = c->io_write_state = CLIENT_IDLE;
+    c->io_parsed_cmd = NULL;
+    c->flag.pending_command = 0;
+
```

Comment:
I know this is the only case we are CURRENTLY using it but might help wrapping it in a dedicated resetClientIOState IMO

Reply:
Done.